app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,70 @@
+require 'date'
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+
+  def edit
+    @task = Task.find(params[:id])
+  end
+
+  def update
+    # task_updates = params[:task]
+    # @task = Task.find(params[:id])
+    # @task.update_attributes(task_updates)
+    # @task.save
+
+    @task = Task.find(params[:id])
+    @task.name = params[:task][:name]
+    @task.description = params[:task][:description]
+    @task.completion_date = params[:task][:completion_date]
+    task_params
+    @task.save
+
+    redirect_to task_path(@task)
+    # redirect_to('/tasks')
+  end
+
+  def show
+    @task = Task.find(params[:id])
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    task = Task.new(
```

Comment:
Because we have the `task_params` helper method we could also have written this as:
```ruby
task = Task.new(task_params)
```